Response-Theory/Coupled-Cluster/RHF/optrot.py

```diff
@@ -0,0 +1,281 @@
+from helper_ccenergy import *
+from helper_cchbar   import *    
+from helper_cclambda import * 
+from helper_ccpert import * 
+
+import psi4
+from psi4 import constants as pc
+
+psi4.set_memory(int(2e9), False)
+psi4.core.set_output_file('output.dat', False)
+
+# can only handle C1 symmetry
+mol = psi4.geometry("""
+ O     -0.028962160801    -0.694396279686    -0.049338350190                                                                  
+ O      0.028962160801     0.694396279686    -0.049338350190                                                                  
+ H      0.350498145881    -0.910645626300     0.783035421467                                                                  
+ H     -0.350498145881     0.910645626300     0.783035421467                                                                  
+symmetry c1        
+""")
+
+# setting up SCF options
+psi4.set_options({'basis': 'sto-3g'})
+psi4.set_options({'scf_type':'PK'})
+psi4.set_options({'e_convergence':1e-10})
+psi4.set_options({'d_convergence':1e-10})
+rhf_e, rhf_wfn = psi4.energy('SCF', return_wfn=True)  
+print('RHF Final Energy                          % 16.10f\n' % rhf_e)
+
+
+# Calculate Ground State CCSD energy
+ccsd = HelperCCEnergy(mol, rhf_e, rhf_wfn, memory=2)
+ccsd.compute_energy(e_conv=1e-10, r_conv=1e-10)
+
+CCSDcorr_E = ccsd.ccsd_corr_e
+CCSD_E = ccsd.ccsd_e
+
+print('\nFinal CCSD correlation energy:          % 16.15f' % CCSDcorr_E)
+print('Total CCSD energy:                      % 16.15f' % CCSD_E)
+
+# Now that we have T1 and T2 amplitudes, we can construct 
+# the pieces of the similarity transformed hamiltonian (Hbar).
+cchbar = HelperCCHbar(ccsd)
+
+# Calculate Lambda amplitudes using Hbar 
+cclambda = HelperCCLambda(ccsd,cchbar)
+cclambda.compute_lambda(r_conv=1e-10)
+
+# frequency of calculation
+omega_nm = 589 
+# convert from nm into hartree
+omega = (pc.c * pc.h * 1e9)/(pc.hartree2J * omega_nm)
+Om   = str(omega)
+Om_0 = str(0)
+
+cart = ['X', 'Y', 'Z']
+pert = {}
+ccpert = {}
+tensor= {}
+cclinresp = {}
+optrot_lg    = np.zeros(9)
+optrot_vg_om = np.zeros(9)
+optrot_vg_0  = np.zeros(9)
+
+###############################################   Length Gauge   ###############################################################
+
+# In length gauge the representation of electric dipole operator is mu i,e. r. So, optical rotation tensor in this gauge 
+# representation can be given by -Im <<mu;L>>, where L is the angular momemtum operator, refer to eq.5 of reference 3.
+# For general form of a response function, refer to eq. 94 of reference 1.
+
+print("\n\n Length Gauge Calculations Starting ..\n\n")
+
+# Obtain the required AO Perturabtion Matrices From Mints
+# Electric Dipole
+dipole_array = ccsd.mints.ao_dipole()
+# Angular Momentum 
+angmom_array = ccsd.mints.ao_angular_momentum()
+
+for i in range(0,3):
+    Mu = "MU_" + cart[i]
+    L = "L_" + cart[i]
+    #P = "P_" + cart[i]
+    # Transform perturbations from AO to MO basis
```

Comment:
Ditto to above can you add some spaces between lines in this file?

Reply:
Done!